### DISCUSSION TITLE
Rule::requiredIf weird behaviour

### DISCUSSION THREAD

**Discussion by mokhosh** [livewire/livewire#3107] [7. Bugs (archived)]
If you have this as a validation rule, you start getting hick ups for some reason:
```php
    protected function rules()
    {
        return [
            'order.name' => ['string', Rule::requiredIf($this->order->service->required_name)],
        ];
    }
```
But if you remove the `requiredIf` rule or use a closure like this, it works perfectly which shouldn't be the case:
```php
    protected function rules()
    {
        return [
            'order.name' => ['string', Rule::requiredIf(fn () => $this->order->service->required_name)],
        ];
    }
```

**Comment by austenc** [comment]
What sort of hiccups are you referring to? Assume you're using the latest version of Livewire?

Please provide as much info as you can, thank you!

**Comment by nathanjansen** [comment]
When are you calling your: `protected function rules()`
And as a follow up; is `$this->order->service` always available?

**Comment by mokhosh** [comment]
@austenc What I'm experiencing is the data changes and gets back with the response but the html is null. But when I change something in the next request what I expected with the previous response comes back, so I'm always one change behind.
Yes latest version.

@nathanjansen I'm not calling it, I assume it's called twice or so by Livewire when hydrating/dehydrating the model.
Yes it's always available.
